Sources/RoutingKit/TrieRouter.swift

```diff
@@ -25,6 +30,14 @@ public final class TrieRouter<Output>: Router, CustomStringConvertible {
     public init(_ type: Output.Type = Output.self, options: Set<ConfigurationOption> = []) {
         self.root = Node()
         self.options = options
+        self.logger = .init(label: "codes.vapor.routingkit")
+    }
+
+    /// Create a new `TrieRouter`.
+    public init(_ type: Output.Type = Output.self, options: Set<ConfigurationOption> = [], logger: Logger) {
+        self.root = Node()
+        self.options = options
+        self.logger = logger
```

Comment:
Why not
```suggestion
    public init(_ type: Output.Type = Output.self, options: Set<ConfigurationOption> = [], logger: Logger = .init(label: "codes.vapor.routingkit")) {
        self.root = Node()
        self.options = options
        self.logger = logger
    }
```

Reply:
Adding a new parameter to an existing method is very slight breaking change that 99.99% noone would even notice.
Gwynne is trying to be careful of that 🙂

It changes the signature of the method, and there will be compiler errors if someone had for some reason used `.map(method)` somewhere.

I plan to do a forums post and complain about this behavior considering it's a lot of trouble for sem-ver-following packages, with minor gains.

Reply:
@MahdiBM Correct, thank you ❤️